db/gorm.go

```diff
@@ -0,0 +1,146 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+var safeNameRE = regexp.MustCompile("^[a-zA-Z0-9_]+$")
+
+// NewMappedSelector returns an object which can be used to automagically query
+// the provided type-mapped database for rows of the parameterized type.
+func NewMappedSelector[T any](executor MappedExecutor) MappedSelector[T] {
+	return &mappedSelector[T]{wrapped: executor}
+}
+
+type mappedSelector[T any] struct {
+	wrapped MappedExecutor
+}
+
+// Query combines the best features of gorp, the go stdlib, and generics.
```

Comment:
```suggestion
// Query performs a SELECT on the appropriate table for T.
// It combines the best features of gorp, the go stdlib, and generics.
```

Note for comparison that https://pkg.go.dev/database/sql#DB.Query says:

> Query executes a query that returns rows, typically a SELECT. The args are for any placeholder parameters in the query.

So it's useful to be explicit that _our_ Query will always do a SELECT.